pkg/services/ngalert/notifier/multiorg_alertmanager.go

```diff
@@ -141,50 +141,13 @@ func (moa *MultiOrgAlertmanager) getLatestConfigs(ctx context.Context) (map[int6
 
 // SyncAlertmanagersForOrgs syncs configuration of the Alertmanager required by each organization.
 func (moa *MultiOrgAlertmanager) SyncAlertmanagersForOrgs(ctx context.Context, orgIDs []int64) {
-	orgsFound := make(map[int64]struct{}, len(orgIDs))
-	dbConfigs, err := moa.getLatestConfigs(ctx)
-	if err != nil {
-		moa.logger.Error("failed to load Alertmanager configurations", "err", err)
-		return
-	}
 	moa.alertmanagersMtx.Lock()
```

Comment:
the unlock is not at the end of the function. We reset lock before stopping the AM that we removed from the map because that can be quite a long operation.